tests/catch/unit/memory/hipFree.cc

```diff
@@ -0,0 +1,452 @@
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-79");
+  return;
+
+  enum StreamType { NULLSTR, CREATEDSTR };
+  auto streamType = GENERATE(NULLSTR, CREATEDSTR);
+  hipStream_t stream{nullptr};
+  if (streamType == CREATEDSTR) HIP_CHECK(hipStreamCreate(&stream));
+
+  hipArray_t arrayPtr{};
+  hipExtent extent{};
+  extent.width = GENERATE(32, 128, 256, 512, 1024);
+  extent.height = GENERATE(0, 32, 128, 256, 512, 1024);
+  hipChannelFormatDesc desc = hipCreateChannelDesc<TestType>();
+
+  HIP_CHECK(hipMallocArray(&arrayPtr, &desc, extent.width, extent.height, hipArrayDefault));
+  lauchLongRunningKernel(50);
+  // make sure device is busy
+  HIP_CHECK_ERROR(hipStreamQuery(nullptr), hipErrorNotReady);
+  // Second free segfaults
+#if HT_AMD
+  SECTION("ArrayDestroy") { HIP_CHECK(workIsDoneCheck<hipArray_t>(arrayPtr, ArrayDestroy)); }
+#endif
+  SECTION("ArrayFree") { HIP_CHECK(workIsDoneCheck<hipArray_t>(arrayPtr, ArrayFree)); }
+
+  if (streamType == CREATEDSTR) HIP_CHECK(hipStreamDestroy(stream));
+}
+
+// On Nvidia devices the CUarray is used when calling hipArrayDestroy
+#if HT_NVIDIA
+TEST_CASE("Unit_hipFreeImplicitSyncArrayD") {
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-81");
+  return;
+  hiparray cuArrayPtr{};
+  CTX_CREATE()
+
+  HIP_ARRAY_DESCRIPTOR cuDesc;
+  cuDesc.Width = 32;
+  cuDesc.Height = 32;
+  cuDesc.Format = HIP_AD_FORMAT_UNSIGNED_INT8;
+  cuDesc.NumChannels = 2;
+  HIP_CHECK(hipArrayCreate(&cuArrayPtr, &cuDesc));
+  lauchLongRunningKernel(50);
+  // make sure device is busy
+  HIP_CHECK_ERROR(hipStreamQuery(nullptr), hipErrorNotReady);
+  SECTION("ArrayDestroy") { HIP_CHECK(workIsDoneCheck<hiparray>(cuArrayPtr, ArrayDestroy)); }
+  CTX_DESTROY()
+}
+#endif
+
+// Freeing a invalid pointer with on device
+TEST_CASE("Unit_hipFreeNegativeDev") {
+  char* devPtr;
+  SECTION("InvalidPtr") {
+    devPtr = new (char);
+    HIP_CHECK_ERROR(freeStuff(devPtr, DevFree), hipErrorInvalidValue);
+    delete devPtr;
+  }
+  SECTION("NullPtr") {
+    devPtr = nullptr;
+    HIP_CHECK(freeStuff(devPtr, DevFree));
+  }
+}
+
+// Freeing a invalid pointer with on device
+TEST_CASE("Unit_hipFreeNegativeHost") {
+#ifdef HT_AMD
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-82");
+  return;
+#endif
+  char* hostPtr{nullptr};
+  SECTION("hipHostRegister") {
+    hostPtr = new char;
+    auto flag = GENERATE(hipHostRegisterDefault, hipHostRegisterPortable, hipHostRegisterMapped);
+    HIP_CHECK(hipHostRegister((void*)hostPtr, sizeof(char), flag));
+    HIP_CHECK_ERROR(freeStuff(hostPtr, HostFree), hipErrorInvalidValue);
+    HIP_CHECK(hipHostUnregister(hostPtr));
+    free(hostPtr);
+  }
+  SECTION("NullPtr") {
+    hostPtr = nullptr;
+    HIP_CHECK(freeStuff(hostPtr, HostFree));
+  }
+  SECTION("InvalidPtr") {
+    hostPtr = new (char);
+    HIP_CHECK_ERROR(freeStuff(hostPtr, HostFree), hipErrorInvalidValue);
+    free(hostPtr);
+  }
+}
+
+// Freeing a invalid pointer with array
+TEST_CASE("Unit_hipFreeNegativeArray") {
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-80&81");
```

Comment:
same question here

Reply:
For this test case we expect the invalid memory to return ```hipErrorInvalidValue``` for invalid pointers being past and ```hipSuccess``` for nullptr, the observed result on AMD devices is the opposite (invalid pointers return success, nullptr retures invalid value)
On Nvidia devices we get get error code 709: (context is destroyed) for arrayFree negative testing and 400 (invalid resource handle) for arrayDestroy which is not the expected behaviour.